pages/blog/graphql-deep-dive-1.mdx

```diff
@@ -0,0 +1,354 @@
+- Setup a [**graphql-config**](https://graphql-config.com/) and this will interface with other tooling like the codegen, VSCode GraphQL extension, GraphQL ESLint and more. This will help a lot since you have one config to manage all the interfacing tools. If you are using the Apollo Stack, you might need an [**apollo-config**](https://www.apollographql.com/docs/devtools/editor-plugins/#setting-up-an-apollo-config) as well
+- If you want to keep your GraphQL code as modular as possible with support for things like dependency injection, try something like [**GraphQL Modules**](https://graphql-modules.com/)
+- Want to interface with multiple different data sources and integrations each with their own format but still have the experience of GraphQL when developing on top of them? Try something like [**GraphQL Mesh**](https://graphql-mesh.com/)
+- Want to use a tool to test GraphQL endpoints? You might need something like [**Insomnia**](https://insomnia.rest/)**,** [**Postman**](https://www.postman.com/)**,** [**Hoppscotch**](https://hoppscotch.io/) **or** [**VSCode REST Client**](https://marketplace.visualstudio.com/items?itemName=humao.rest-client)
+
+And while I can talk more about this, it will never end cause the ecosystem is too huge and thriving.
+
+**I use REACT/Angular /Vue/Web Components. How do I integrate GraphQL with my components?**
+
+Again, the front end ecosystem is huge as well with its own set of tooling and libraries.
+
+In my case, I typically try to work on the frontend without any framework (I use [Lit Elements](https://lit-element.polymer-project.org/) in my case and we will have a separate blog on that soon), the tool you use completely depends on your requirements here.
+
+- Apollo Client does have a good integration with these frameworks including [React](https://www.apollographql.com/docs/react/), [iOS](https://www.apollographql.com/docs/ios/) and [**Android**](https://www.apollographql.com/docs/android/) — so, you might want to check that out
+- Using Vue? You can try [**Vue Apollo**](https://apollo.vuejs.org/guide/)
+- Using web components with Apollo Stack for GQL? You might want to check out [**Apollo Elements**](https://github.com/apollo-elements/apollo-elements)
+- Using vanilla JS or TS or using web components and want to have a framework-independent way of doing things? You can stick to the GraphQL codegen itself since it takes care of almost everything underneath. Or if you want, you can also use Apollo Client’s vanilla version `@apollo/client/core`.
+  [**Apollo Elements**](https://github.com/apollo-elements/apollo-elements) does come with support for a lot of webcomponent libraries like [Lit](https://lit-element.polymer-project.org/), [Fast](https://www.fast.design/) and [Gluon](https://github.com/ruphin/gluonjs) or even without any of it and hence is quite flexible.
+
+Well, there are lot of other ways we haven’t talked about and this is just a start.
```

Comment:
I would also mention here Relay and URQL, or maybe even just a simple react-query or graphql-request

Reply:
Sure. Will add it.